### ISSUE TITLE
No meshes are rendered in 3D samples, but GUI works [Arch Linux]

### ISSUE THREAD

**Issue by UberLambda** [openframeworks/openFrameworks#5422]
When I try to run any sample in the 3d/ category the GUI is rendered correctly; any meshes, however, are not. This happens both from a freshly-compiled master build and `v20170106_linux64gcc`. The samples are run via `make RunRelease`.

Screenshot of 3DPrimitivesExample for reference:
![screenshot](https://cloud.githubusercontent.com/assets/7994377/22177886/fab12bec-e027-11e6-9361-81bb98da5533.png)
* * * * *
GPU: AMD RX480, managed by the `amdgpu` driver.

```
$ glxinfo -B
name of display: :0
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: X.Org (0x1002)
    Device: AMD POLARIS10 (DRM 3.3.0 / 4.8.13-1-ARCH, LLVM 3.9.1) (0x67df)
    Version: 13.0.3
    Accelerated: yes
    Video memory: 8150MB
    Unified memory: no
    Preferred profile: core (0x1)
    Max core profile version: 4.3
    Max compat profile version: 3.0
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.1
OpenGL vendor string: X.Org
OpenGL renderer string: Gallium 0.4 on AMD POLARIS10 (DRM 3.3.0 / 4.8.13-1-ARCH, LLVM 3.9.1)
OpenGL core profile version string: 4.3 (Core Profile) Mesa 13.0.3
OpenGL core profile shading language version string: 4.30
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 3.0 Mesa 13.0.3
OpenGL shading language version string: 1.30
OpenGL context flags: (none)

OpenGL ES profile version string: OpenGL ES 3.1 Mesa 13.0.3
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.10
```
* * * * *
Should be unrelated, but the samples all emit the following errors when run:
```
[notice ] ofGstVideoGrabber: Probing devices with udev...
[ error ] ofGstVideoGrabber: initGrabber(): no devices found, exiting without initializing
[warning] ofGstVideoUtils: update(): ofGstVideoUtils not loaded
```

**Comment by arturoc** [comment]
can you post the output of `glxinfo | grep direct_state_access`

**Comment by UberLambda** [comment]
```
$ glxinfo | grep direct_state_access
    GL_ARB_derivative_control, GL_ARB_direct_state_access, 
```

**Comment by hamoid** [comment]
Here's mine (Laptop with Intel Graphics 4600):

        GL_ARB_depth_clamp, GL_ARB_derivative_control, GL_ARB_direct_state_access,

**Comment by arturoc** [comment]
hey thanks, can you test if my latest commit to master fixes it for you?

**Comment by hamoid** [comment]
It does work for me with your latest commit to master 👍

**Comment by arturoc** [comment]
can you test again? the last commit wasn't working on the test servers since they have an older version of some libraries

**Comment by UberLambda** [comment]
`master` is working fine 👍

**Comment by arturoc** [comment]
great thanks for testing!

**Comment by hamoid** [comment]
This seems to be happening again after a system update, as mentioned at https://forum.openframeworks.cc/t/my-vbos-are-invisible-probably-because-of-a-system-update/26176/1

```
$glxinfo -B
name of display: :0
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Intel Open Source Technology Center (0x8086)
    Device: Mesa DRI Intel(R) Haswell Mobile  (0x416)
    Version: 17.0.1
    Accelerated: yes
    Video memory: 1536MB
    Unified memory: yes
    Preferred profile: core (0x1)
    Max core profile version: 4.5
    Max compat profile version: 3.0
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.1
OpenGL vendor string: Intel Open Source Technology Center
OpenGL renderer string: Mesa DRI Intel(R) Haswell Mobile 
OpenGL core profile version string: 4.5 (Core Profile) Mesa 17.0.1
OpenGL core profile shading language version string: 4.50
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 3.0 Mesa 17.0.1
OpenGL shading language version string: 1.30
OpenGL context flags: (none)

OpenGL ES profile version string: OpenGL ES 3.1 Mesa 17.0.1
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.10

```

GL_ARB_direct_state_access is still there. I can post the whole `glxinfo` output if useful.

**Comment by arturoc** [comment]
what happens if you change all the if (GLEW_EXT_direct_state_access) { in ofBufferObject.cpp with if(false)?

**Comment by hamoid** [comment]
In that case it works :)

**Comment by hamoid** [comment]
Gee... the ```if(false)``` fix stopped working :(

I'm back at the blank screen as shown at the top. I tried ofBufferObject.cpp from December, from master, and modified as you suggested here. None of them produce triangles.

```
$glxinfo -B
name of display: :0
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Intel Open Source Technology Center (0x8086)
    Device: Mesa DRI Intel(R) Haswell Mobile  (0x416)
    Version: 17.1.0
    Accelerated: yes
    Video memory: 1536MB
    Unified memory: yes
    Preferred profile: core (0x1)
    Max core profile version: 4.5
    Max compat profile version: 3.0
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.1
OpenGL vendor string: Intel Open Source Technology Center
OpenGL renderer string: Mesa DRI Intel(R) Haswell Mobile 
OpenGL core profile version string: 4.5 (Core Profile) Mesa 17.1.0
OpenGL core profile shading language version string: 4.50
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 3.0 Mesa 17.1.0
OpenGL shading language version string: 1.30
OpenGL context flags: (none)

OpenGL ES profile version string: OpenGL ES 3.1 Mesa 17.1.0
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.10
```

**Comment by hamoid** [comment]
Ok mesa-17.0.5 works, mesa-17.1.0 doesn't. I'll just keep 17.0.5 until there's an update.

**Comment by d-egan** [comment]
I had the same problem on v20170714_linux64gcc6_nightly but I fixed it by using the [ofGLProgrammableRenderer](http://openframeworks.cc/documentation/gl/ofGLProgrammableRenderer/), with `settings.setGLVersion();` Worked with 4.5 and 3.0.  

```
$ glxinfo -B
name of display: :0
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Intel Open Source Technology Center (0x8086)
    Device: Mesa DRI Intel(R) Haswell Mobile  (0x416)
    Version: 17.3.0
    Accelerated: yes
    Video memory: 1536MB
    Unified memory: yes
    Preferred profile: core (0x1)
    Max core profile version: 4.5
    Max compat profile version: 3.0
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.1
OpenGL vendor string: Intel Open Source Technology Center
OpenGL renderer string: Mesa DRI Intel(R) Haswell Mobile 
OpenGL core profile version string: 4.5 (Core Profile) Mesa 17.3.0
OpenGL core profile shading language version string: 4.50
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 3.0 Mesa 17.3.0
OpenGL shading language version string: 1.30
OpenGL context flags: (none)

OpenGL ES profile version string: OpenGL ES 3.1 Mesa 17.3.0
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.10
```

**Comment by arturoc** [comment]
huh, that's interesting, thanks for reporting it. @hamoid can you test if that works for you too?

**Comment by hamoid** [comment]
1. Normally I have my OF patched replacing 8 instances of

    if (GLEW_ARB_direct_state_access) {

with 

    if (false && GLEW_ARB_direct_state_access) {

I just disabled that patch and faces and wireframe disappeared from the example, as it usually happens with my integrated Intel.

2. Next I replaced

    ofSetupOpenGL(1280, 720, OF_WINDOW);
    ofRunApp(new ofApp());

with

    ofGLWindowSettings settings;
    settings.setGLVersion(4, 5);
    ofCreateWindow(settings);
    ofRunApp(new ofApp());

which made faces and triangles visible again! But notice the rendering difference:

Default:

![2017-12-21-130156_1280x720_scrot](https://user-images.githubusercontent.com/108264/34255278-41226918-e650-11e7-9021-d7e62e801288.png)

OpenGL 4.5:

![2017-12-21-130021_1024x768_scrot](https://user-images.githubusercontent.com/108264/34255270-351d7ebe-e650-11e7-950f-fbd0cd5fa45f.png)

Much smoother, but notice the "glitch" on top-right highlight (there's a sharp cut).
